[sw/benchmarks] fix coremark benchmark nightly test

timothytrippel opened the pull request:
The coremark benchmark test meson build target was commented out in
PR #12206, since it failed to build when the OTTF entry point was
renamed from `main` to `_ottf_main` (the `ottf_start.asm` assembly is
used with the benchmark code to setup the device). However, this benchmark
is run as a nightly regression in DV simulation. As a result of the missing
build target, this test was failing.

This patches the coremark benchmark code to work with the new entry
point symbol: `_ottf_main`.

Signed-off-by: Timothy Trippel <ttrippel@google.com>

cindychip reviewed (commented):
Thanks for fixing this! I do not know how vendor patches work, will leave for other sw experts to approve it :)

mcy reviewed (commented):
Instead of doing this, maybe we should port coremark to Bazel in the same way I did freertos? Meson is going away soon so this will be mooted.

timothytrippel commented:
Yeah, I agree, we should port coremark to bazel in the same way, I added a issue about this: #12504.

Since it was such a simple fix to fix meson, < 3min, I went ahead and fixed it for the DV team so that there wouldn't be a failing test on the dashboard util we can get coremark bazelified.
